boot/boot_test.go

```diff
@@ -910,6 +909,15 @@ func (s *bootenv20Suite) TestCoreParticipant20SetNextNewKernelSnapWithReseal(c *
 	c.Assert(err, IsNil)
 	c.Assert(rebootRequired, Equals, true)
 
+	// make sure the env was updated
+	bvars, err := tab.GetBootVars("kernel_status", "snap_kernel", "snap_try_kernel")
+	c.Assert(err, IsNil)
+	c.Assert(bvars, DeepEquals, map[string]string{
+		"kernel_status":   boot.TryStatus,
+		"snap_kernel":     s.kern1.Filename(),
+		"snap_try_kernel": s.kern2.Filename(),
```

Comment:
as discussed on IRC, this is not quite what we want to be testing to match what's used in practice today (well this test itself is valid too, just nothing uses this kind of setup right now), because MockTrustedAssetBooloader is an EnvRefKernelBootloader, not an ExtractedRunKernelImageBootloader and we should have tests for TrustedAssetBootloader that is also an ExtractedRunKernelImageBootloader

to be clear, today, the only trusted assets bootloader we have is grub, which is not an EnvRefKernelBootloader, it is an ExtractedRunKernelImageBootloader, but when we add support for lk, etc. we may have a trusted asset bootloader that is also an env ref kernel bootloader.